### DISCUSSION TITLE
Issue when updating Xcode version: Cache not invalidated

### DISCUSSION THREAD

**Discussion by dcacenabes** [swiftyfinch/Rugby#393] [Q&A] [answered]
We are in the process of bumping our Xcode version from 15.3 to 15.4. We are also bumping the SDK we are using to build from iOS 17.4 to iOS 17.5
As part of this process we are getting the errors when trying to build the app. This is our build command:
```
set -o pipefail && env NSUnbufferedIO=YES xcodebuild -workspace OurApp.xcworkspace -scheme OurAppScheme -derivedDataPath /Users/xx/Library/Developer/Xcode/DerivedData/OurApp-clgerggzsgdgwjajqxghgtfiaxvu -sdk 'iphonesimulator17.5' -destination 'platform=iOS Simulator,id=50958E2B-7506-4CB9-8363-1D6F97F09F85' -resultBundlePath './fastlane/test_output/OurApp.xcresult' -retry-tests-on-failure -test-iterations 5 test | tee '/Users/xx/Library/Logs/scan/Glovo-OurAppScheme.log' | xcbeautify -qq --is-ci
```
And this is the type of error we get.
```
Error: cannot load module 'XXAPI' built with SDK 'iphonesimulator17.4' when using SDK 'iphonesimulator17.5': /Users/xxxx/.rugby/bin/XXAPI/DebugStaging-iphonesimulator-arm64/506e39e/XXAPI.framework/Modules/XXAPI.swiftmodule/arm64-apple-ios-simulator.swiftmodule
import XXAPI
```

I was reviewing the Rugby logs, and I found that it found a match for all binaries:
`✓ Found 100% Binaries (205)`

I do not know if this is expected behavior or not, but what I was expecting was to get 0 hits. Also, I could not find a way in the docs to tell Rugby which is the SDK it should use for building. As a workaround, I could delete the whole caches, but this can be tricky to coordinate as we have multiple people working with different Xcode versions.
Ideally, I would like to be able to specify to Rugby which version of the iOS SDK it should be using for building and for checking checksums, and I would also expect the tool to mark cache as missed if the SDK version is not the same.

Also, I do not know if this can be of any help, but if I run xcodebuild command for our app without specifying the SDK version (simply adding `-sdk iphonesimulator` instead of `-sdk iphonesimulator17.5`) it will fallback to the one defined by the current Xcode version.

Looking forward to hearing from you!

David

**Comment by dcacenabes** [comment]
I took a look at the source code, and I found that for the sdk parameter Rugby [only specifies](https://github.com/swiftyfinch/Rugby/blob/68ef7c1903815203308779714145c9d0d39bea46/Sources/RugbyFoundation/Core/Build/XcodeBuild/XcodeBuild.swift#L122) 
`"-sdk \(options.sdk.xcodebuild)",` 
which [can only be iphonesimulator or iphoneos](https://github.com/swiftyfinch/Rugby/blob/68ef7c1903815203308779714145c9d0d39bea46/Sources/RugbyFoundation/Core/Build/XcodeBuild/XcodeBuild.swift#L52)

**Comment by swiftyfinch** [comment]
@dcacenabes Hello!

Hm, maybe it would be enough to save the `xcode_version` in addition to the `swift_version`:
https://github.com/swiftyfinch/Rugby/blob/main/Sources/RugbyFoundation/Core/Common/Hashers/TargetsHasher.swift#L63

You can try to use this provider:
https://github.com/swiftyfinch/Rugby/blob/main/Sources/RugbyFoundation/Core/Env/XcodeCLTVersionProvider.swift#L38

**Reply by swiftyfinch** [reply]
Let me know if you need some help

**Reply by dcacenabes** [reply]
Thanks! Gonna take a look 👀 👨‍💻

**Reply by dcacenabes** [reply]
I am working on the fix. I wanted to update existing tests and write new ones where it might apply, but I am getting compilation errors (only from Xcode). Created a separate discussion to review it:
https://github.com/swiftyfinch/Rugby/discussions/394

**Reply by dcacenabes** [reply, marked as answer]
I have created a draft PR https://github.com/swiftyfinch/Rugby/pull/395

Looking forward to hearing your thoughts and happy to apply any feedback you might drop!

**Reply by swiftyfinch** [reply]
@dcacenabes 🏈 **[v2.10.0](https://github.com/swiftyfinch/Rugby/releases/tag/2.10.0)**

**Reply by dcacenabes** [reply]
We've already bumped our pipelines to 2.10.0 and everything is working as expected.

**Reply by swiftyfinch** [reply]
@dcacenabes That's great!
